src/app/shared/services/api/api.service.ts

```diff
@@ -0,0 +1,79 @@
+import {inject, Injectable} from '@angular/core';
+import { DocumentReference} from "@angular/fire/compat/firestore";
+import {ITrip} from "../../../../models/types";
+import {
+  Firestore,
+  collection,
+  doc,
+  collectionData,
+  docData,
+  addDoc,
+  deleteDoc,
+  updateDoc
+} from "@angular/fire/firestore";
+import {from, Observable} from "rxjs";
+
+import {IItem} from "../../../models/user";
+
+@Injectable({
+  providedIn: 'root'
+})
+export class ApiService {
+  firestore: Firestore = inject(Firestore);
+  constructor(
+  ) {}
+
+  public getAllTrips(): Observable<ITrip[]> {
+    const tripCollection = collection(this.firestore, 'trips');
+    return collectionData(tripCollection) as Observable<ITrip[]>;
+  }
+
+  getTripInfo(tripId: string): Observable<ITrip>{
+    const tripReference = doc(this.firestore, `trips/${tripId}/`);
+    return docData(tripReference) as Observable<ITrip>;
+  }
+
+  addTrip(trip: ITrip): Observable<DocumentReference<ITrip>>{
+    const tripCollection = collection(this.firestore, 'trips');
+    return from(addDoc(tripCollection, trip)) as unknown as Observable<DocumentReference<ITrip>>;
+  }
+
+  addItineraryItem(item: IItem, tripId: string): Observable<DocumentReference<IItem>> {
+    const itemCollection = collection(this.firestore, `trips/${tripId}/itinerary`);
+    return from( addDoc(itemCollection, item) ) as unknown as Observable<DocumentReference<IItem>>;
+  }
```

Comment:
Unknown type conversion before converting to Observable. Please review and assert types properly